src/arrayMethodJoin.js

```diff
@@ -4,8 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joined = '';
+
+    for (const item of this) {
```

Comment:
```suggestion
    for (const i = 0;i < this.length; i++) {
```
что б дальше не использовать this.indexOf(item) для проверки.